<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>960</width>
    <height>680</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ServeDebug æ™ºèƒ½åµŒå…¥å¼è°ƒè¯•ç»ˆç«¯</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* ========== å…¨å±€æ ·å¼è¡¨ ========== */

/* 1. åŸºç¡€è‰²è°ƒå®šä¹‰ */
QWidget {
        background-color: #f5f6fa; /* æ•´ä½“èƒŒæ™¯ç° */
        color: #2f3640;            /* å…¨å±€æ–‡å­—æ·±ç° */
        font-family: &quot;Segoe UI&quot;, &quot;Microsoft YaHei&quot;, sans-serif;
        font-size: 10pt;
}

/* 2. å·¦ä¾§ä¾§è¾¹æ é£æ ¼ */
QFrame#frameSidebar {
        background-color: #ffffff;
        border-right: 1px solid #dcdde1;
}

/* 3. æŒ‰é’®é€šç”¨é£æ ¼ */
QPushButton {
        border: 1px solid #dcdde1;
        border-radius: 6px;
        background-color: #f5f6fa;
        padding: 6px 12px;
        min-height: 20px;
}
QPushButton:hover {
        background-color: #e1e4ea;
        border-color: #c4c7cc;
}
QPushButton:pressed {
        background-color: #d1d4db;
        padding-top: 7px; /* æŒ‰ä¸‹ä½ç§»æ„Ÿ */
        padding-left: 13px;
}

/* 4. ä¸»è¦æ“ä½œæŒ‰é’®ï¼ˆè¿æ¥ï¼‰- è“è‰²é«˜äº® */
QPushButton#btnConnect {
        background-color: #0097e6;
        color: white;
        border: 1px solid #0084c9;
        font-weight: bold;
        font-size: 11pt;
        padding: 8px;
}
QPushButton#btnConnect:hover {
        background-color: #00a8ff;
}
QPushButton#btnConnect:pressed {
        background-color: #0084c9;
}
QPushButton#btnConnect:disabled {
        background-color: #bdc3c7;
        border-color: #bdc3c7;
}

/* 5. å±é™©æ“ä½œæŒ‰é’®ï¼ˆæ–­å¼€ï¼‰- çº¢è‰²é«˜äº® */
QPushButton#btnDisconnect {
        background-color: #e84118;
        color: white;
        border: 1px solid #c23616;
        padding: 8px;
}
QPushButton#btnDisconnect:hover {
        background-color: #ff4d4d;
}
QPushButton#btnDisconnect:pressed {
        background-color: #c23616;
}

/* 6. è¾“å…¥æ§ä»¶é£æ ¼ */
QComboBox, QLineEdit {
        border: 1px solid #dcdde1;
        border-radius: 4px;
        padding: 4px 8px;
        background-color: white;
        selection-background-color: #0097e6;
}
QComboBox:focus, QLineEdit:focus {
        border: 1px solid #0097e6;
}
QComboBox::drop-down {
        border: none;
        width: 20px;
}

/* 7. è®¾å¤‡åˆ—è¡¨é£æ ¼ */
QListWidget {
        border: 1px solid #dcdde1;
        border-radius: 4px;
        background-color: #ffffff;
        outline: none;
}
QListWidget::item {
        height: 32px;
        padding-left: 4px;
        border-bottom: 1px solid #f0f0f0;
}
QListWidget::item:selected {
        background-color: #eaf4fd;
        color: #0097e6;
        border-left: 3px solid #0097e6;
}
QListWidget::item:hover:!selected {
        background-color: #f5f6fa;
}

/* 8. æ—¥å¿—ç»ˆç«¯é£æ ¼ */
QPlainTextEdit#textLog {
        background-color: #000000; /* æ·±è‰²èƒŒæ™¯ */
        color: #ffffff;
        border-radius: 6px;
        font-family: &quot;Consolas&quot;, &quot;Courier New&quot;, monospace;
        font-size: 11pt;
        selection-background-color: #34ace0;
        selection-color: white;
}

/* 9. æ»šåŠ¨æ¡ç¾åŒ– */
QScrollBar:vertical {
        border: none;
        background: #f1f1f1;
        width: 10px;
        margin: 0px;
}
QScrollBar::handle:vertical {
        background: #bdc3c7;
        min-height: 20px;
        border-radius: 5px;
}
QScrollBar::handle:vertical:hover {
        background: #95a5a6;
}
QScrollBar::add-line:vertical, QScrollBar::sub-line:vertical {
        height: 0px;
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_Main">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QFrame" name="frameSidebar">
      <property name="minimumSize">
       <size>
        <width>260</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_Sidebar">
       <property name="spacing">
        <number>15</number>
       </property>
       <property name="leftMargin">
        <number>15</number>
       </property>
       <property name="topMargin">
        <number>20</number>
       </property>
       <property name="rightMargin">
        <number>15</number>
       </property>
       <property name="bottomMargin">
        <number>20</number>
       </property>
       <item>
        <widget class="QLabel" name="labelTitle">
         <property name="font">
          <font>
           <family>Segoe UI</family>
           <pointsize>10</pointsize>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="styleSheet">
          <string notr="true">color: #2f3640;</string>
         </property>
         <property name="text">
          <string>é…ç½®é¢æ¿</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupSettings">
         <property name="title">
          <string>é€šä¿¡å‚æ•°</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <property name="verticalSpacing">
           <number>10</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="labelBaud">
            <property name="text">
             <string>æ³¢ç‰¹ç‡</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="cbBaudRate">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="currentText">
             <string>9600</string>
            </property>
            <item>
             <property name="text">
              <string>9600</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>19200</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>38400</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>57600</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>115200</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QCheckBox" name="chkHexDisplay">
            <property name="text">
             <string>Hex 16è¿›åˆ¶æ˜¾ç¤º</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupDevices">
         <property name="title">
          <string>è®¾å¤‡é€‰æ‹©</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_Devices">
          <property name="spacing">
           <number>8</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>6</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="layoutDeviceHeader">
            <item>
             <widget class="QLabel" name="labelDeviceCount">
              <property name="styleSheet">
               <string notr="true">color: #7f8c8d; font-size: 9pt;</string>
              </property>
              <property name="text">
               <string>æ‰«æä¸­...</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_Devices">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="btnManualRefresh">
              <property name="maximumSize">
               <size>
                <width>60</width>
                <height>24</height>
               </size>
              </property>
              <property name="text">
               <string>åˆ·æ–°</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QListWidget" name="listDevices">
            <property name="selectionBehavior">
             <enum>QAbstractItemView::SelectRows</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="layoutStatus">
         <property name="spacing">
          <number>5</number>
         </property>
         <item>
          <widget class="QLabel" name="labelStatus">
           <property name="font">
            <font>
             <family>Segoe UI</family>
             <pointsize>10</pointsize>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">color: #e84118;</string>
           </property>
           <property name="text">
            <string>ğŸ”´ æœªè¿æ¥</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="labelCurrentPort">
           <property name="styleSheet">
            <string notr="true">color: #7f8c8d; font-size: 9pt;</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_Sidebar">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="btnConnect">
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>å¼€å§‹è¿æ¥</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnDisconnect">
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="text">
          <string>æ–­å¼€è¿æ¥</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widgetContent" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_Content">
       <property name="spacing">
        <number>10</number>
       </property>
       <property name="leftMargin">
        <number>15</number>
       </property>
       <property name="topMargin">
        <number>15</number>
       </property>
       <property name="rightMargin">
        <number>15</number>
       </property>
       <property name="bottomMargin">
        <number>15</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_LogTools">
         <item>
          <widget class="QLabel" name="labelLogTitle">
           <property name="font">
            <font>
             <family>Segoe UI</family>
             <pointsize>10</pointsize>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>æ•°æ®ç›‘æ§ç»ˆç«¯</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_Log">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="btnClearLog">
           <property name="maximumSize">
            <size>
             <width>80</width>
             <height>28</height>
            </size>
           </property>
           <property name="text">
            <string>æ¸…ç©ºæ—¥å¿—</string>
           </property>
           <property name="iconSize">
            <size>
             <width>12</width>
             <height>12</height>
            </size>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="textLog">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="layout_Send">
         <property name="spacing">
          <number>10</number>
         </property>
         <item>
          <widget class="QLineEdit" name="txtSend">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>35</height>
            </size>
           </property>
           <property name="placeholderText">
            <string>è¯·è¾“å…¥å¾…å‘é€çš„æ•°æ® (ASCII)...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnSend">
           <property name="minimumSize">
            <size>
             <width>90</width>
             <height>34</height>
            </size>
           </property>
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="styleSheet">
            <string notr="true">
             background-color: #2f3542;
             color: white;
             font-weight: bold;
             border: 1px solid #2f3542;
            </string>
           </property>
           <property name="text">
            <string>å‘é€</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
